checker/src/main/java/org/checkerframework/checker/resourceleak/ResourceLeakVisitor.java

```diff
@@ -0,0 +1,226 @@
+      AnnotationMirror createsMustCallFor, MustCallAnnotatedTypeFactory mcAtf) {
+    return AnnotationUtils.getElementValue(
+        createsMustCallFor, mcAtf.getCreatesMustCallForValueElement(), String.class, "this");
+  }
+
+  /**
+   * Returns all the literal strings present in the @CreatesMustCallFor annotations on the given
+   * element. This version correctly handles multiple CreatesMustCallFor annotations on the same
+   * element. This differs from {@link
+   * org.checkerframework.checker.mustcall.CreatesMustCallForElementSupplier#getCreatesMustCallForExpressions(MethodInvocationNode,
+   * GenericAnnotatedTypeFactory, CreatesMustCallForElementSupplier)} in that this version does not
+   * take into account the calling context when parsing the strings; instead, the literal values
+   * written by the programmer are returned.
+   *
+   * @param elt an executable element
+   * @param mcAtf a MustCallAnnotatedTypeFactory, to source the value element
+   * @param atypeFactory a ResourceLeakAnnotatedTypeFactory
+   * @return the literal strings present in the @CreatesMustCallFor annotation(s) of that element,
+   *     substituting the default "this" for empty annotations. This method returns the empty list
+   *     iff there are no @CreatesMustCallFor annotations on elt. The returned list is always
+   *     modifiable if it is non-empty.
+   */
+  /*package-private*/ static List<String> getLiteralCreatesMustCallForValues(
+      ExecutableElement elt,
+      MustCallAnnotatedTypeFactory mcAtf,
+      ResourceLeakAnnotatedTypeFactory atypeFactory) {
+    AnnotationMirror createsMustCallForList =
+        atypeFactory.getDeclAnnotation(elt, CreatesMustCallFor.List.class);
+    List<String> result = new ArrayList<>(4);
+    if (createsMustCallForList != null) {
+      List<AnnotationMirror> createsMustCallFors =
+          AnnotationUtils.getElementValueArray(
+              createsMustCallForList,
+              mcAtf.getCreatesMustCallForListValueElement(),
+              AnnotationMirror.class);
+      for (AnnotationMirror cmcf : createsMustCallFors) {
+        result.add(getLiteralCreatesMustCallForValue(cmcf, mcAtf));
+      }
+    }
+    AnnotationMirror createsMustCallFor =
+        atypeFactory.getDeclAnnotation(elt, CreatesMustCallFor.class);
+    if (createsMustCallFor != null) {
+      result.add(getLiteralCreatesMustCallForValue(createsMustCallFor, mcAtf));
+    }
+    return result;
+  }
+
+  @Override
+  public Void visitVariable(VariableTree node, Void p) {
+    Element varElement = TreeUtils.elementFromTree(node);
+
+    if (varElement.getKind().isField()
+        && !checker.hasOption(MustCallChecker.NO_LIGHTWEIGHT_OWNERSHIP)
+        && rlTypeFactory.getDeclAnnotation(varElement, Owning.class) != null) {
+      checkOwningField(varElement);
+    }
+
+    return super.visitVariable(node, p);
+  }
+
+  /**
+   * Checks validity of a final field {@code field} with an {@code @Owning} annotation. Say the type
```

Comment:
How do we know the field is final?  The caller does not seem to check that, nor does the body of this method.

Reply:
This was an error - this method was originally written when we only supported `final` owning fields, but when we generalized it this documentation must not have been updated. I've removed the reference to the field being final.